logical/translate_response.go

```diff
@@ -35,6 +36,34 @@ func SanitizeResponse(input *Response) *HTTPResponse {
 		}
 	}
 
+	return httpResp
+}
+
+func HTTPResponseToLogicalResponse(input *HTTPResponse) *Response {
+	logicalResp := &Response{
+		Data:     input.Data,
+		warnings: input.Warnings,
+	}
+
+	if input.LeaseID != "" {
+		logicalResp.Secret = &Secret{
+			LeaseID: input.LeaseID,
+		}
+		logicalResp.Secret.Renewable = input.Renewable
```

Comment:
Those correspond to the embedded `LeaseOptions` and while you *can* initialize that at the same time as the containing object, I just...didn't feel like doing it that way.  :-)